src/controllers/restful/PaymentController.ts

```diff
@@ -4,216 +4,207 @@
+     * It will be generated here after the first interaction
+     * started by BackendApp (checkPaymentToPagoPa)
+     * For the next messages, BackendApp will provide the same codiceContestoPagamento
+     */
+    const codiceContestoPagamento = generateCodiceContestoPagamento();
 
-  /**
-   * Convert the input provided by BackendApp (RESTful request)
-   * to a PagoPa request (SOAP request), mapping useful information
-   * Some static information will be obtained by pagoPaConfig, to identify this client
-   * If something wrong into input will be detected during mapping, and error will be provided as response
-   */
-  const errorOrPaymentCheckRequestPagoPa = PaymentsConverter.getPaymentsCheckRequestPagoPa(
-    pagoPaConfig,
-    errorOrPaymentsCheckRequest.value,
-    codiceContestoPagamento
-  );
-  if (errorOrPaymentCheckRequestPagoPa.isLeft()) {
-    return left(
-      RestfulUtils.sendErrorResponse(
-        res,
-        ControllerError.ERROR_INVALID_INPUT,
-        HttpErrorStatusCode.keys.BAD_REQUEST
-      )
+    // Convert the input provided by BackendApp (RESTful request) to a PagoPa request (SOAP request).
+    // Some static information will be obtained by pagoPaConfig, to identify this client.
+    const errorOrPaymentCheckRequestPagoPa = PaymentsConverter.getPaymentsCheckRequestPagoPa(
+      pagoPaConfig,
+      paymentCheckRequest,
+      codiceContestoPagamento
     );
-  }
-
-  // Send the SOAP request to PagoPa (VerificaRPT message)
-  const errorOrPaymentCheckPagoPaResponse = await PaymentsService.sendPaymentCheckRequestToPagoPa(
-    errorOrPaymentCheckRequestPagoPa.value,
-    paymentVerificaRPTPagoPaClient
-  );
-  if (errorOrPaymentCheckPagoPaResponse.isLeft()) {
-    RestfulUtils.sendUnavailableAPIError(res);
-    return left(ControllerError.ERROR_API_UNAVAILABLE);
-  }
+    if (isLeft(errorOrPaymentCheckRequestPagoPa)) {
+      const error = errorOrPaymentCheckRequestPagoPa.value;
+      return ResponseErrorValidation(
+        "Invalid payment check request",
+        error.message
+      );
+    }
+    const paymentCheckRequestPagoPa = errorOrPaymentCheckRequestPagoPa.value;
 
-  /**
-   * Check PagoPa response content.
-   * If it contains an error, an HTTP error will be provided to BackendApp
-   */
-  if (
-    errorOrPaymentCheckPagoPaResponse.value.nodoVerificaRPTRisposta.esito ===
-    PPTPortTypes.Esito.KO
-  ) {
-    return left(
-      RestfulUtils.sendErrorResponse(
-        res,
-        ControllerError.REQUEST_REJECTED,
-        HttpErrorStatusCode.keys.BAD_REQUEST
-      )
+    // Send the SOAP request to PagoPa (VerificaRPT message)
+    const errorOrPaymentCheckPagoPaResponse = await PaymentsService.sendPaymentCheckRequestToPagoPa(
+      paymentCheckRequestPagoPa,
+      paymentVerificaRPTPagoPaClient
     );
-  }
+    if (isLeft(errorOrPaymentCheckPagoPaResponse)) {
+      const error = errorOrPaymentCheckPagoPaResponse.value;
+      return ResponseErrorInternal(
```

Comment:
we use standard responses from italia-ts-commons package (no need for custom responses in RestfulUtils.*)